### PR TITLE
fix(calendar-input)

### PR DESCRIPTION
**Pull request by Heymdall**
Blur event on input come before calendar focus, and calendar-input fire blur event.

### CODE REVIEW

File: src/calendar/calendar.jsx [hunk 1]
```diff
@@ -327,7 +327,9 @@ class Calendar extends React.Component {
             clearTimeout(this.blurTimeoutId);
         }
 
+        console.log('calendar blur');
```
Comment (teryaew) [inline src/calendar/calendar.jsx]: ?

File: src/calendar-input/calendar-input.jsx [hunk 2]
```diff
@@ -565,7 +567,16 @@ class CalendarInput extends React.Component {
             ...focusedState
         };
 
-        let newFocused = newState.isInputFocused || newState.isCalendarFocused;
+        // Blur event come before focus, so we may found that input already loose focus and calendar don't get it yet.
+        // So we need to check what element will receive focus, and if it inside calendar - keep current focused state.
+        const relatedTarget = event.relatedTarget || // FF doesn't add related target to blur events
```
Comment (teryaew) [inline src/calendar-input/calendar-input.jsx]: Why do we mix `let` & `const`?

Comment (Heymdall) [inline src/calendar-input/calendar-input.jsx]: Why do we use `let`? I will change it for now, I think we should discuss it

File: src/calendar/calendar.jsx (hunk 1 above)
Comment (Heymdall) [inline src/calendar/calendar.jsx]: fixed

File: src/calendar-input/calendar-input.jsx [hunk 3]
```diff
@@ -565,7 +567,16 @@ class CalendarInput extends React.Component {
             ...focusedState
         };
 
-        let newFocused = newState.isInputFocused || newState.isCalendarFocused;
+        // Blur event come before focus, so we may found that input already loose focus and calendar don't get it yet.
```
Comment (Heymdall) [inline src/calendar-input/calendar-input.jsx]: fixed

**Pull request merged**